### PR TITLE
Add contract tests for empty/missing Location headers

### PR DESCRIPTION
**Pull request by cwaldren-ld**
While implementing redirect handling for the Rust eventsource-client, I needed to implement logic to handle missing/empty Location headers.

It would be undesirable for a client to enter an infinite loop if presented with `Location: ` or no Location header at all. Therefore, these tests assert that the client returns an error in both cases.

I put it behind a capability flag because I'm not sure that all clients implement this defensive functionality. 

The Go eventsource client does pass these tests, and specifically for 301s does even have a helpful error string:
```
2022/03/16 15:00:44.710944 [HTTP behavior/client handles empty Location header with 301 status]: Failed to start stream: Get "http://localhost:8111/endpoints/156": 301 response missing Location header
```
(for 307s it just says `error 307`)

### CODE REVIEW

File: ssetests/tests_http_behavior.go [hunk 1]
```diff
@@ -41,6 +41,30 @@ func DoHTTPBehaviorTests(t *ldtest.T) {
...
 		})
+
+		// The intention of these tests are to ensure the client does not, when presented
+		// with an empty or missing Location header:
+		// 1) Loop infinitely
+		// 2) Keep using the current URL without emitting an error
+		for _, action := range []string{"empty", "missing"} {
+			t.Run(fmt.Sprintf("client handles %s Location header with %d status", action, status), func(t *ldtest.T) {
+				t.RequireCapability("empty-or-missing-location")
+
```
Comment (cwaldren-ld) [inline ssetests/tests_http_behavior.go]: > empty-or-missing-location

Not the greatest of names.

**Comment by cwaldren-ld** [comment]
Made test mandatory.

**Review by eli-darkly — APPROVED** [review]
